### PR TITLE
Prevents users from building a BulkProcessor with a null client

### PR DESCRIPTION
**Pull request by oyiadom**
I suggest updating the BulkProcessor so that it prevents users from building a BulkProcessor with a null client.  If the client is null, then the class should throw an exception with a message that describes the cause of the exception.  Earlier today, when I passed a null client to the builder() method, later received an exception, and attempted to get the exception's message in my afterBulk() listener (e.g. e.getMessage()), the message was null.  It took me a while to pinpoint the cause of the problem.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java [hunk 1]
```diff
@@ -145,6 +145,10 @@ public BulkProcessor build() {
     }
 
     public static Builder builder(Client client, Listener listener) {
+        if (client == null) {
+            throw new Exception("The client you specified while building a BulkProcessor is null");
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java]: Can you make it a NullPointerException or an IllegalArgumentException instead?

**Comment by oyiadom** [comment]
@jpountz Done.  I went with a NullPointerException.

File: core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java (hunk 1 above)
Comment (oyiadom) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java]: @jpountz Done. I went with a NullPointerException.

**Comment by jpountz** [comment]
Thanks @oyiadom

**Pull request merged**